qlib/rl/order_execution/simulator_qlib.py

```diff
@@ -65,22 +80,32 @@ def reset(
             strategy=strategy_config,
             executor=executor_config,
             benchmark=order.stock_id,
-            account=1e12,
+            account=cash_limit if cash_limit is not None else int(1e12),
             exchange_kwargs=exchange_config,
-            pos_type="InfPosition",
+            pos_type="Position" if cash_limit is not None else "InfPosition",
         )
 
         assert isinstance(self._executor, NestedExecutor)
 
+        self.report_dict: dict = {}
+        self.decisions: List[BaseTradeDecision] = []
         self._collect_data_loop = collect_data_loop(
             start_time=order.date,
             end_time=order.date,
             trade_strategy=strategy,
             trade_executor=self._executor,
+            return_value=self.report_dict,
         )
         assert isinstance(self._collect_data_loop, Generator)
 
-        self._last_yielded_saoe_strategy = self._iter_strategy(action=None)
+        if backtest_mode:
+            executor: BaseExecutor = self._executor
```

Comment:
Add comments that it should be removed in the future

Reply:
Done~